### PR TITLE
feat: 🎸 Support filtering by node name in rich-text-links

### PR DESCRIPTION
**Pull request by Silhoue**
Needed in the GraphQL API.

### CODE REVIEW

File: packages/rich-text-links/src/index.ts [hunk 1]
```diff
@@ -29,13 +29,13 @@ function getRichTextEntityLinks(document: Document): EntityLinks {
...
-function addLinksFromRichTextNode(node: Node, links: EntityLinkMaps): void {
+function addLinksFromRichTextNode(node: Node, links: EntityLinkMaps, type?: string): void {
   const toCrawl: Node[] = [node];
 
   while (toCrawl.length > 0) {
-    const { data, content } = toCrawl.pop() as Block;
+    const { data, content, nodeType } = toCrawl.pop() as Block;
 
-    if (isLinkObject(data)) {
+    if ((!type || nodeType === type) && isLinkObject(data)) {
```
Comment (sbezludny) [inline packages/rich-text-links/src/index.ts]: please extract variable to clarify the purpose of this check.

File: packages/rich-text-links/src/index.ts [hunk 2]
```diff
@@ -12,15 +12,15 @@ export type EntityLinkNodeData = {
 /**
  * Given a rich text document, returns all entity links.
  */
-function getRichTextEntityLinks(document: Document): EntityLinks {
+function getRichTextEntityLinks(document: Document, type?: string): EntityLinks {
```
Comment (sbezludny) [inline packages/rich-text-links/src/index.ts]: May I ask you to annotate a new argument

File: packages/rich-text-links/src/index.ts (hunk 1 above)
Comment (Silhoue) [reply packages/rich-text-links/src/index.ts]: done

File: packages/rich-text-links/src/index.ts (hunk 2 above)
Comment (Silhoue) [reply packages/rich-text-links/src/index.ts]: What do you mean by "annotate a new argument"? Can you show me some example?

Comment (sbezludny) [reply packages/rich-text-links/src/index.ts]: Sorry for the confusion, I meant something like this:
```js
/**
*  Extracts entity links from a Rich Text document.
*/
function getRichTextEntityLinks(
/**
*  An instance of a Rich Text Document.
*/
document: Document, 
/**
* Node type. Will be returned entity links only with given type.
*/
type?: string): EntityLinks
```

Comment (Silhoue) [reply packages/rich-text-links/src/index.ts]: I thought TypeScript annotations 😅  Updated.

**Pull request merged**